runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerEnvironmentFactory.java

```diff
@@ -114,20 +112,16 @@ public RemoteEnvironment createEnvironment(Environment environment) throws Excep
     String workerId = idGenerator.getId();
 
     // Prepare docker invocation.
-    Path workerPersistentDirectory = Files.createTempDirectory("worker_persistent_directory");
-    Path semiPersistentDirectory = Files.createTempDirectory("semi_persistent_dir");
```

Comment:
Why are we removing these mount points?

Reply:
They break on mac and they're not useful atm. They are for caching to achieve faster restarts, but we generate new directories every time. It's also an optional feature anyway.

Reply:
CC: @robertwb

Reply:
Ah, I know what it is. I thought I had fixed this here, but perhaps I only did it in the hacking branch. We need to explicitly specify the `/tmp` prefix here, so it doesn't default to `/var/tmp/...`.

Basically, we need to do it the same way as in https://github.com/bsidhom/beam/blob/3e9b24a1247033e5d75f80e106d3d75052389799/runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/SingletonDockerEnvironmentManager.java#L90

Reply:
I think it's worth doing this here rather than removing the mount altogether because we will eventually want it.

Reply:
I think we should omit this till if/when we decide to do it right. The current code was a bit confused (the path inside the container shouldn't be created on the host) and if we use kubernetes, say, then it would be done in a different way. For local runs, having no mounts is not unreasonable.

Reply:
That's just used as a proxy for generating a "random" directory name inside of the container. We can remove that and manually generate a random (or fixed) container directory name if you don't like the side effect of creating one on the host. I've added https://github.com/apache/beam/pull/5820 to address this, so feel free to keep the deletions here.